disruption_py/core/physics_method/runner.py

```diff
@@ -237,19 +232,17 @@ def populate_method(
                 raise
 
         physics_method_params.logger.warning(
-            "[Shot %s]: Failed to populate %s with error %s",
-            physics_method_params.shot_id,
-            name,
-            e,
+            "{name}: Failed! {e}",
+            name=name,
+            e=e,
         )
-        physics_method_params.logger.debug("%s", traceback.format_exc())
+        physics_method_params.logger.opt(exception=True).debug(e)
         result = {col: [np.nan] for col in bound_method_metadata.columns}
 
     physics_method_params.logger.info(
-        "[Shot %s]: Completed %s, time_elapsed: %s",
-        physics_method_params.shot_id,
-        name,
-        time.time() - start_time,
+        "{t:.3f}s : {name}",
+        name=name,
+        t=time.time() - start_time,
```

Comment:
how come `start_time` is passed as an arg, rather than getting created as a first thing by the method itself?

Reply:
Here, the start time means the start time of running the physics method at the level of the shot. We log the elapsed time since the start of getting data for the shot at the completion of each physics method.

Here's an alternate way to structure these time logs:
- Each shot logs the time to grab all data for the shot in our that shot's success/warning/error log eg `#1150805014 | Success! Most parameters have data: 62/63 (98.41%) in 1s`
- Each physics method logs only the time it took to complete that one physics method

Let me know how that sounds

Reply:
okay, we'll go through this tomorrow! thanks.